### DISCUSSION TITLE
Distinguish between null and undefined with parse()

### DISCUSSION THREAD

**Discussion by gee4vee** [eemeli/yaml#436] [Q&A] [answered]
Is there a way I can configure `parse` so that I can distinguish between a `value` not having a value and having `null` as the value? As it stands, in either case I get `null` as the value.

**Comment by eemeli** [comment, marked as answer]
With the default schema (YAML 1.2 core), an empty string and `null` are both representations of a null value.

If you use the `parseDocument` API, the parsed node's `source` does provide this info:

```js
YAML.parseDocument('null').contents
```

```js
Scalar {
  value: null,
  range: [ 0, 4, 4 ],
  source: 'null',
  type: 'PLAIN'
}
```

**Reply by gee4vee** [reply]
So I'm looking at the parsed document contents but seeing `null` as the value for a yaml that has nothing (not empty string). I'm looking to distinguish no value (not empty string) and `null`.

![image](https://user-images.githubusercontent.com/179807/212953418-363febcd-2c2f-48ea-b425-64ef4ab52543.png)

**Reply by gee4vee** [reply]
I suppose I have to look at the source, type, and value to determine if it's empty string, nothing, or `null`.
